core/dnsserver/address.go

```diff
@@ -94,3 +94,82 @@ const (
 	TransportTLS  = "tls"
 	TransportGRPC = "grpc"
 )
+
+const keyPartsSeparator = "##"
+
+type addrKey struct {
```

Comment:
Why do we need this and can't extend zoneAddr (which has almost the same bits?)

Reply:
That was my first idea .. but ZoneAddr is used to compute/populate the Keys of ServerBloc.
and we did not wanted Address there.

I preferred to make a separated small object used only for the overlapping check. and has not the constrainst of the Keys.